packages/menu/src/types.ts

```diff
@@ -0,0 +1,150 @@
+ *
+ * Use of this source code is governed under the Apache License, Version 2.0
+ * found at http://www.apache.org/licenses/LICENSE-2.0.
+ */
+
+import { ButtonHTMLAttributes, HTMLProps, ReactNode, RefObject } from 'react';
+
+export type ItemValue = string | number;
+
+export interface ISelectedItem {
+  value: ItemValue;
+  label?: string;
+  name?: string;
+  type?: 'radio' | 'checkbox';
+}
+
+export interface IMenuItemBase extends ISelectedItem {
+  disabled?: boolean;
+}
+
+export type MenuItemSeparator = {
+  value: ItemValue;
+  separator: boolean;
+};
+
+export type MenuItemGroup = {
+  label: string;
+  items: (IMenuItemBase | MenuItemSeparator)[];
+};
+
+export type MenuItem = IMenuItemBase | MenuItemSeparator | MenuItemGroup;
+
+export type MenuChangeHandlerArgs = {
+  type: string;
+  isExpanded?: boolean;
+  selectedItems?: ISelectedItem[];
+  focusedValue?: ItemValue;
+};
+
+export interface IUseMenuProps<T = HTMLElement, L = HTMLElement> {
+  /**
+   * Provides an ordered list of menu items
+   *
+   * @param {ItemValue} item.value Unique item value
+   * @param {string} item.label Human-readable item display value
+   * @param {string} item.name A shared name corresponding to an item radio group.
+   * @param {boolean} item.disabled Indicates the item is not interactive
+   * @param {boolean} item.separator Indicates the item is a placeholder for a separator.
+   * @param {IMenuItemBase[]} item.items Groups a list of items
+   */
+  items: MenuItem[];
+  /** Provides ref access to the underlying trigger element */
+  triggerRef: RefObject<T>;
+  /** Provides ref access to the underlying menu element */
+  menuRef: RefObject<L>;
+  /** Indicates that the element is not interactive */
+  disabled?: boolean;
+  /** Prefixes IDs for the menu */
+  idPrefix?: string;
+  /** Sets the expansion in a controlled menu */
+  isExpanded?: boolean;
+  /** Determines default expansion in an uncontrolled menu */
+  defaultExpanded?: ItemValue;
+  /** Determines menu expansion on menu initialization */
+  initialExpanded?: boolean;
+  /** Sets the focused value in a controlled menu */
+  focusedValue?: ItemValue;
+  /** Determines default focused value in an uncontrolled menu */
+  defaultFocusedValue?: ItemValue;
+  /** Determines focused value expansion on menu initialization */
+  initialFocusedValue?: ItemValue;
+  /** Sets the selected values in a controlled menu */
+  selectedItems?: ISelectedItem[];
+  /**
+   * Handles menu state changes
+   *
+   * @param {string} changes.type The event type that triggered the change
+   * @param {boolean} [changes.isExpanded] The updated menu expansion
+   * @param {ISelectedItem[]} [changes.selectedItems] The updated selection values
+   * @param {ItemValue} [changes.focusedValue] The updated focused value
+   */
+  onChange?: (changes: MenuChangeHandlerArgs) => void;
+  /** Sets the environment where the menu is rendered */
+  environment?: Window;
+}
+
+export interface IUseMenuReturnValue<T, L> {
```

Comment:
Can't figure out why I get `LegacyRef` vs `RefObject` issues here without making the interface generic, but a nearly identical structure in Combobox doesn't. Open to ideas/pairing to get to the bottom of it if it feels unnecessary.